### PR TITLE
Razorpay integration

### PR DESCRIPTION
**Pull request by mohammedfabinsha**

### CODE REVIEW

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx [hunk 1]
```diff
@@ -73,19 +121,124 @@ const CheckoutView: React.FC = () => {
...
+      } else {
+        //const shipmentOrderId = ....create shipment orderId
+        const codData = {
+          orderId,
+          amount: amountInPaise,
+          currency,
+        };
+        const payId = await createCODTrigger({
+          ...codData,
+        });
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:239]: Please move this to libs/razorpay/swr

File: libs/razorpay/src/services/create-razorpay-order/service.ts [hunk 2]
```diff
@@ -12,26 +12,35 @@ export async function createRazorpayOrder(data: CreateRPOrderRequest) {
...
+        amount: validatedData.amount,
+        status: resp.status,
+      },
+    });
+    const response = {
+      razorpayOrderId: resp.id,
+      paymentId: newPayment.id,
+    };
+    return response;
+  } else return null;
```
Comment (fahimalizain) [inline libs/razorpay/src/services/create-razorpay-order/service.ts:45]: Raise error here

File: libs/razorpay/src/services/razorpay-webhook/service.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  const secret = '6d2Lv76AugGojJGt1Wa7MEgFZmSZaq5z';
+
+  const webhookSignature = data.signature;
+  delete (data as { signature?: string }).signature;
+
+  const dataString = JSON.stringify(data);
+
+  const generatedSignature = CryptoJS.HmacSHA256(dataString, secret).toString(
+    CryptoJS.enc.Hex,
+  );
```
Comment (fahimalizain) [inline libs/razorpay/src/services/razorpay-webhook/service.ts:18]: Extract this out to libs/razorpay/src/services/signature.ts
Copy can be found at libs/razorpay/src/services/verify-signature/service.ts

File: vestido/vestido-storefront/app/api/orders/[orderSlug]/cod/route.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { ZodError } from 'zod';
```
Comment (fahimalizain) [inline vestido/vestido-storefront/app/api/orders/[orderSlug]/cod/route.ts:1]: - Deprecate dedicated COD Route.
- When paymentType is COD, insert OrderPayment Entry within CreateOrder Endpoint itself.

File: vestido/vestido-storefront/app/api/payment/[paymentSlug]/route.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import { ZodError } from 'zod';
```
Comment (fahimalizain) [inline vestido/vestido-storefront/app/api/payment/[paymentSlug]/route.ts:1]: Change paymentSlug to paymentId

Comment (fahimalizain) [inline vestido/vestido-storefront/app/api/payment/[paymentSlug]/route.ts:1]: Change `/api/payment/[paymentId]` to `/api/payments/[paymentId]`. Please pluralize

File: vestido/vestido-storefront/app/api/orders/[orderSlug]/payment/route.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import { ZodError } from 'zod';
```
Comment (fahimalizain) [inline vestido/vestido-storefront/app/api/orders/[orderSlug]/payment/route.ts:2]: Please change `POST /api/orders/[orderId]/payment` to `POST /api/orders/[orderId]/payments`

File: vestido/vestido-storefront/app/api/orders/[orderSlug]/payment/route.ts [hunk 7]
```diff
@@ -1,5 +1,6 @@
...
```
Comment (fahimalizain) [inline vestido/vestido-storefront/app/api/orders/[orderSlug]/payment/route.ts:1]: Change [orderSlug] to [orderId]

File: libs/razorpay/src/services/verify-signature/service.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import * as CryptoJS from 'crypto-js';
```
Comment (fahimalizain) [inline libs/razorpay/src/services/verify-signature/service.ts:1]: Rename this service to verify-payment

**Review by fahimalizain — CHANGES_REQUESTED** [review]

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx (hunk 1 above)
Comment (mohammedfabinsha) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx]: Done

File: libs/razorpay/src/services/create-razorpay-order/service.ts (hunk 2 above)
Comment (mohammedfabinsha) [reply libs/razorpay/src/services/create-razorpay-order/service.ts]: returned error here

File: libs/razorpay/src/services/razorpay-webhook/service.ts (hunk 3 above)
Comment (mohammedfabinsha) [reply libs/razorpay/src/services/razorpay-webhook/service.ts]: Done

File: vestido/vestido-storefront/app/api/orders/[orderSlug]/cod/route.ts (hunk 4 above)
Comment (mohammedfabinsha) [reply vestido/vestido-storefront/app/api/orders/[orderSlug]/cod/route.ts]: Done

File: libs/razorpay/src/services/verify-signature/service.ts (hunk 8 above)
Comment (mohammedfabinsha) [reply libs/razorpay/src/services/verify-signature/service.ts]: changed to process-payment as verify-payment already exist in swr

**Review by fahimalizain — APPROVED** [review]

**Pull request merged**